features/update/self/feature.java

```diff
@@ -0,0 +1,73 @@
+package update.self;
+
+import io.temporal.activity.Activity;
+import io.temporal.activity.ActivityInterface;
+import io.temporal.activity.ActivityMethod;
+import io.temporal.client.WorkflowClient;
+import io.temporal.sdkfeatures.Assertions;
+import io.temporal.sdkfeatures.Feature;
+import io.temporal.sdkfeatures.Run;
+import io.temporal.sdkfeatures.Runner;
+import io.temporal.workflow.UpdateMethod;
+import update.updateutil.UpdateUtil;
+
+import java.time.Duration;
+import java.util.Objects;
+
+@ActivityInterface
+public interface feature extends Feature, SelfUpdateWorkflow {
+
+  String INITIAL_STATE = "Not signaled";
+
+  String FINAL_STATE = "Signaled!";
+
+  @ActivityMethod
+  void selfUpdate();
+
+  @UpdateMethod()
+  void update();
+
+
+  class Impl implements feature {
+    private String state = INITIAL_STATE;
```

Comment:
Sometimes you put a blank line between class decl and fields, sometimes not (I have not been applying very close scrutiny to PRs on this repo, but I am worried things are gonna get a bit bad...I wonder if we can apply a format check in CI?)